roles/openshift_health_checker/openshift_checks/etcd_traffic.py

```diff
@@ -0,0 +1,72 @@
+"""Module that scans journalctl for messages caused as a symptom of increased etcd traffic"""
+
+from openshift_checks import OpenShiftCheck, OpenShiftCheckException, get_var
+
+
+class EtcdTraffic(OpenShiftCheck):
+    """Check if host is being affected by an increase in etcd traffic."""
+
+    name = "etcd_traffic"
+    tags = ["health", "etcd"]
+
+    @classmethod
+    def is_active(cls, task_vars):
+        """Skip hosts that do not have etcd or masters in their group names."""
+        group_names = get_var(task_vars, "group_names", default=[])
+        valid_group_names = "masters" in group_names or "etcd" in group_names
+
+        version = get_var(task_vars, "openshift", "common", "version", default="0.0.0")
+        parsed_version = float(".".join(version.split(".")[0:2]))
+        valid_version = parsed_version == 3.4 or parsed_version == 3.5
+
+        return super(EtcdTraffic, cls).is_active(task_vars) and valid_group_names and valid_version
+
+    def run(self, tmp, task_vars):
+        master_count_int = self._get_master_count(task_vars)
+        deploy_type = get_var(task_vars, "openshift", "common", "service_type")
+        unit = self._get_systemd_unit(master_count_int, deploy_type)
+
+        log_matchers = [{
+            "start_regexp": r"Starting Etcd Server",
+            "regexp": r"etcd: sync duration of [^,], expected less than 1s",
+            "unit": unit
+        }]
+
+        match = self.module_executor("search_journalctl", {
+            "log_matchers": log_matchers,
+        }, task_vars)
+
+        if match.get("matched"):
+            msg = ("Higher than normal etcd traffic detected. "
+                   "OpenShift 3.4 introduced an increase in etcd traffic by at least a factor of 4."
+                   "\nUpgrading to OpenShift 3.6 is recommended in order to fix this issue."
+                   "Please refer to https://access.redhat.com/solutions/2916381 for more information.")
+            return {"failed": True, "msg": msg}
+
+        if match["failed"]:
+            return {"failed": True, "msg": match["msg"]}
+
+        return {}
+
+    @staticmethod
+    def _get_master_count(task_vars):
+        master_count = get_var(task_vars, "openshift", "master", "master_count", default=1)
+        try:
+            master_count_int = int(master_count)
+        except ValueError:
+            msg = ("Non-numeric value ({}) found for 'openshift.master.master_count' in ansible_facts."
+                   "Make sure an integer value for this variable has been set in your inventory.")
```

Comment:
Was not sure how to word this message. Maybe just leave the suggestion out and simply state that `master_count` was a non-numeric value?